internal/api/v2/handler_transfer_initiations_create.go

```diff
@@ -0,0 +1,170 @@
+package v2
+
+import (
+	"encoding/json"
+	"errors"
+	"math/big"
+	"net/http"
+	"time"
+
+	"github.com/formancehq/go-libs/api"
+	"github.com/formancehq/go-libs/pointer"
+	"github.com/formancehq/payments/internal/api/backend"
+	"github.com/formancehq/payments/internal/models"
+	"github.com/formancehq/payments/internal/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type createTransferInitiationRequest struct {
+	Reference            string            `json:"reference"`
+	ScheduledAt          time.Time         `json:"scheduledAt"`
+	Description          string            `json:"description"`
+	SourceAccountID      string            `json:"sourceAccountID"`
+	DestinationAccountID string            `json:"destinationAccountID"`
+	ConnectorID          string            `json:"connectorID"`
+	Provider             string            `json:"provider"`
+	Type                 string            `json:"type"`
+	Amount               *big.Int          `json:"amount"`
+	Asset                string            `json:"asset"`
+	Validated            bool              `json:"validated"`
+	Metadata             map[string]string `json:"metadata"`
+}
+
+func (r *createTransferInitiationRequest) Validate() error {
+	if r.Reference == "" {
+		return errors.New("reference is required")
+	}
+
+	if r.SourceAccountID != "" {
+		_, err := models.AccountIDFromString(r.SourceAccountID)
+		if err != nil {
+			return err
+		}
+	}
+
+	if r.DestinationAccountID != "" {
+		_, err := models.AccountIDFromString(r.DestinationAccountID)
+		if err != nil {
+			return err
+		}
+	}
+
+	_, err := models.PaymentInitiationTypeFromString(r.Type)
+	if err != nil {
+		return err
+	}
+
+	if r.Amount == nil {
+		return errors.New("amount is required")
+	}
+
+	if r.Asset == "" {
+		return errors.New("asset is required")
+	}
+
+	return nil
+}
+
+func transferInitiationsCreate(backend backend.Backend) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		ctx, span := otel.Tracer().Start(r.Context(), "v2_transferInitiationsCreate")
+		defer span.End()
+
+		payload := createTransferInitiationRequest{}
+		if err := json.NewDecoder(r.Body).Decode(&payload); err != nil {
+			otel.RecordError(span, err)
+			api.BadRequest(w, ErrMissingOrInvalidBody, err)
+			return
+		}
+
+		setSpanAttributesFromRequest(span, payload)
+
+		if err := payload.Validate(); err != nil {
+			otel.RecordError(span, err)
```

Comment:
Could be implemented in another ticket, but we should probably implement the tracing as http middleware so that whoever implements a new handler doesn't have to remember to call these tracing functions every time.

Reply:
You're right, I added another ticket to fix this !